src/Raven.Client/Http/RequestExecuter.cs

```diff
@@ -122,10 +122,25 @@ private void UpdateTopologyCallback(object _)
             }
         }
 
-        public async Task ExecuteAsync<TResult>(JsonOperationContext context, RavenCommand<TResult> command)
+        public void Execute<TResult>(RavenCommand<TResult> command)
         {
             var choosenNode = ChooseNodeForRequest(command);
-            await ExecuteAsync(choosenNode, context, command);
+            JsonOperationContext context;
+            using (_contextPoll.AllocateOperationContext(out context))
```

Comment:
Why aren't you calling `Task ExecuteAsync<TResult>(RavenCommand<TResult> command)` here?